drivers/wifi/nxp/nxp_wifi_drv.c

```diff
@@ -276,20 +276,20 @@ int nxp_wifi_wlan_event_callback(enum wlan_event_reason reason, void *data)
 	case WLAN_REASON_UAP_CLIENT_CONN:
 		wlan_get_current_uap_network(&nxp_wlan_network);
 #ifdef CONFIG_NXP_WIFI_11AX
-			if (nxp_wlan_network.dot11ax) {
-				ap_sta_info.link_mode = WIFI_6;
-			}
+		if (nxp_wlan_network.dot11ax) {
```

Comment:
This isn't a blocking request given the timeline for the release, but could we do something like the following to make this code more readable?

```
if (COND_CODE_1(CONFIG_NXP_WIFI_11AX, nxp_wlan_network.dot11ax, 0)) {
    ap_sta_info.link_mode = WIFI_6;
} else if (COND_CODE_1(CONFIG_NXP_WIFI_11AX, nxp_wlan_network.dot11ac, 0)) {
    ap_sta_info.link_mode = WIFI_5;
} else if (nxp_wlan_network.dot11n) {
    ap_sta_info.link_mode = WIFI_4;
} else {
    ap_sta_info.link_mode = WIFI_3;
}
```

COND_CODE_1 will allow us to only use the `dot11ax` or `dot11ac` fields if the Kconfig is enabled, since it seems that if the Kconfig is on the field won't be defined in the structure.

Reply:
Good idea! 
After that, I can modify all similar places together.